Form checkbox and radio button components (#3198)
canonical-web-and-design/vanilla-framework: templates/docs/base/forms.md

### Checkbox

<span class="p-label--new">New</span>

Use the checkbox component to select one or more options. To provide fully featured Vanilla style and behaviour of the checkbox a specific markup structure is needed around the checkbox input (see example below).

To disable the checkbox component a `disabled` property needs to be added on respective `<input type='checkbox'>` element and `.is-disabled` class needs to be used on the component root `.p-checkbox` element.

<div class="embedded-example"><a href="/docs/examples/patterns/forms/checkbox/" class="js-example">
View example of the checkbox components
</a></div>

#### Inline checkbox

> To use checkbox component in padded containers (table cells, some list items) use the `.p-checkbox--inline` variant of the component to make sure the checkbox and the label text are properly aligned with other inline text.

Reviewer: ```suggestion
When placing the checkbox component in padded containers (table cells, some list items), use the `.p-checkbox--inline` variant. It ensures the checkbox and the label text are properly aligned with other inline text.
```

<div class="embedded-example"><a href="/docs/examples/patterns/forms/checkbox-inline/" class="js-example">
View example of the inline checkbox components
</a></div>

#### Heading checkbox

> To use checkbox component in headings use the `.p-checkbox--heading` variant of the component to make sure the checkbox and the label text are properly aligned.

Reviewer: ```suggestion
To style the checkbox label as a heading, use the `.p-checkbox--heading` variant of the component. It ensures correct vertically alignement of the checkbox and the heading.
```

Author: `--heading` variant doesn't style the label as heading, only make sure checkbox is aligned, so skipping this.

This applies also to all modifier classes on checkbox elements, such as `is-h1`, `is-h2`, `is-h3`, `is-h4`, `is-h5`, `is-muted-heading`, `is-muted-inline-heading`, `is-inline-label`, or `is-table-header`.

Please refer to the [tick elements comparison example](/docs/examples/patterns/forms/tick-comparison) to find equivalent usage of new `.p-checkbox` component.

### Radio button

<span class="p-label--new">New</span>

Use radio buttons to select one of the given set of options. To provide fully featured Vanilla style and behaviour of the radio button a specific markup structure is needed around the radio button input (see example below).

To disable the radio component a `disabled` property needs to be added on respective `<input type='radio'>` element and `.is-disabled` class needs to be used on the component root `.p-radio` element.

<div class="embedded-example"><a href="/docs/examples/patterns/forms/radio/" class="js-example">
View example of the radio components
</a></div>

#### Inline radio button

> To use radio component in padded containers (table cells, some list items) use the `.p-radio--inline` variant of the component to make sure the radio button and the label text are properly aligned with other inline text.

Reviewer: ```suggestion
When placing the radio component in padded containers (table cells, some list items), use the `.p-radio--inline` variant of the component to make sure the radio button and the label text are properly aligned with other inline text.
```

<div class="embedded-example"><a href="/docs/examples/patterns/forms/radio-inline/" class="js-example">
View example of the inline radio components
</a></div>

#### Heading radio button

> To use radio component in headings use the `.p-checkbox--heading` variant of the component to make sure the radio button and the label text are properly aligned.

Reviewer: ```suggestion
To style the radio label as a heading `.p-radio--heading` variant of the component to make sure the radio button and the label text are properly aligned.
```

Author: `--heading` variant doesn't style the label as heading, only make sure radio is aligned, so skipping this.
